src/content_cao.cpp

```diff
@@ -1465,9 +1466,18 @@ void GenericCAO::updateAnimation()
 {
 	if(m_animated_meshnode == NULL)
 		return;
-	m_animated_meshnode->setFrameLoop(m_animation_range.X, m_animation_range.Y);
-	m_animated_meshnode->setAnimationSpeed(m_animation_speed);
+
+	if (m_animated_meshnode->getStartFrame() != m_animation_range.X ||
```

Comment:
Yes: It stops the animation from restarting if multiple animation packages are received at once by the client. Basically it doesn't re-set the animation if the parameters are exactly the same, which has been an annoyance I didn't notice and fix when I initially implemented the code.